### ISSUE TITLE
2.x changes

### ISSUE THREAD

**Issue by pmk65** [json-editor/json-editor#450]
### 2.x changes

- Editors refactored to remove redundant code and make output/functionality more uniform.
- New standardized options format for plugin based editors. Options can be set globally or in schema.
- Editors can now use JavaScript callback functions in options.
- Select2 & Selectize is no longer applied globally to all elements, but can be enabled for a single field. (You can now use both Select2 and Selectize fields on the same form)
- All editors now respects the "compact" option and displays headers, description & infoText correctly _(Output is Theme controlled, so might not work/look correctly with old 1.x Themes)_.

#### New Theme format (Old themes can still be used, but might not display correctly in some cases.):
- Themes can now define Style Rules and use those instead of Inline Styles.
- Themes have own options for changing various aspects of the display.
- Functions added for creating radio, checkbox, button, range and other previously hadcoded elements.

#### Themes in 2.x format:
- Spectre (New Theme, created by @pmk65)
- Tailwind (Updated Theme, created by @mirannda)
**_Note: We still need someone to update/cleanup the Bootstrap and Foundation themes._** 

#### Refactored editors:
- **String** - Ace, SCEditor & SimpleMDE code moved to separate files that extends string editor.
- **Select** - Select2 & Selectize code moved to separate files that extends select editor.
- **Radio** - Output now created from Theme functions instead of hardcoded. Now extends select editor instead of string editor. Supports vertical and horizontal layout.
- **Multiselect** - Select2 & Selectize code moved to separate files that extends multiselect editor.

#### New Editors:
- **Select2** - Extracted from select editor. 
- **Selectize** - Extracted from select editor.
- **Ace** - Extracted from string editor.
- **SCEditor** - Extracted from string editor.
- **SimpleMDE** - Extracted from string editor.
- **Jodit** - WYSIWYG Editor, using [Jodit v.3](https://xdsoft.net/jodit/)
- **Autocomplete** - Autocomplete/Autosuggest using [Autocomplete](https://autocomplete.trevoreyre.com/#/) 
- **IP** - IPv4/IPv6 and hostname fields with validation.

#### New special editors: (Does not leave any footprint in form results)
- **Button** - Add buttons with JavaScript callbacks, to form.
- **Info** - Insert textual blocks in form.

Also various minor bugs and issues have been fixed during refactoring/cleanup.

**Comment by schmunk42** [comment]
Let's put that into the CHANGELOG in `devleop/2.x`

**Comment by pmk65** [comment]
I have also updated several of the [Playground](https://pmk65.github.io/jedemov2/dist/demo.html) examples and added new ones too. (So now some of the examples REQUIRE the 2.x version)

**Comment by sirockin** [comment]
I forked develop/2.x at commit 72fc61eca7cea08d9361f9fed34a5cff390fe152 to work on the webpack stuff. Hoping to have something for you in a few days. From what I can see of your changes since then, it shouldn't be too hard for us to merge.

**Comment by sirockin** [comment]
Now submitted as https://github.com/json-editor/json-editor/pull/486

**Comment by sirockin** [comment]
Is there a proposed timetable for merging/releasing this branch? I have some other issues I'd like to work on and I'm wondering whether it would be better to wait until then.

**Comment by schmunk42** [comment]
@sirockin Issue #486 can be merged ASAP. I'd recommend to continue with `develop/2.x`, we had no big changes on `master`.

**Comment by pmk65** [comment]
Generally all develop/2.x will/should be merged as soon as it has passed tests and reviews.
(@schmunk42 Shouldn't we upgrade @sirockin to project collaborator?)

**Comment by schmunk42** [comment]
@sirockin Would you be OK with it?

**Comment by sirockin** [comment]
No problem - would it confer any responsibilities? - I ask because I have big chunks of time when I'm offline so can't always be available.

**Comment by pmk65** [comment]
No. it's just easier to assign issues to you and you have the ability to review and merge PRs.

**Comment by schmunk42** [comment]
I've sent you invitations to the developer group.
